src/debug/ee/debugger.cpp

```diff
@@ -15354,6 +15354,10 @@ HRESULT Debugger::FuncEvalSetup(DebuggerIPCE_FuncEvalInfo *pEvalInfo,
         _ASSERTE(filterContext != NULL);
 
         ::SetIP(filterContext, (UINT_PTR)GetEEFuncEntryPoint(::FuncEvalHijack));
+        // The SP must be aligned at call boundaries, but not necessarily on every assembly instruction
+        // Since we are hijacking the thread at an arbitrary location, we must align the SP.
+        // Align the stack down to STACK_ALIGN_SIZE
+        ::SetSP(filterContext, ALIGN_DOWN(::GetSP(filterContext), STACK_ALIGN_SIZE));
```

Comment:
I am not familiar with the code, but I wonder - don't we need to fix the SP back to its original value after the funceval somehow? We are getting a context pointer to the Thread::m_debuggerFilterContext and I wonder if that context is not used for other stuff in the debugger too.

Reply:
I was wondering the same thing myself all weekend. I think that's why I added you as a reviewer...

I have abandoned this approach.